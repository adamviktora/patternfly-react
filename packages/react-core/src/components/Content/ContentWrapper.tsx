import * as React from 'react';
import styles from '@patternfly/react-styles/css/components/Content/content';
import { css } from '@patternfly/react-styles';

export interface ContentWrapperProps extends React.HTMLProps<HTMLDivElement> {
  /** Content rendered within the ContentWrapper */
  children?: React.ReactNode;
  /** Additional classes added to the ContentWrapper */
  className?: string;
  /** Flag to indicate the all links in the content block have visited styles applied if the browser determines the link has been visited */
  isVisited?: boolean;
}

export const ContentWrapper: React.FunctionComponent<ContentWrapperProps> = ({
  children,
  className = '',
  isVisited = false,
  ...props
}: ContentWrapperProps) => (
  <div {...props} className={css(styles.content, isVisited && styles.modifiers.visited, className)}>
    {children}
  </div>
);

ContentWrapper.displayName = 'ContentWrapper';
