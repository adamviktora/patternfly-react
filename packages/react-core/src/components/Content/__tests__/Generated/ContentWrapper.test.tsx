/**
 * This test was generated
 */
import * as React from 'react';
import { render } from '@testing-library/react';
import { ContentWrapper } from '../../ContentWrapper';
// any missing imports can usually be resolved by adding them here
import {} from '../..';

it('ContentWrapper should match snapshot (auto-generated)', () => {
  const { asFragment } = render(<ContentWrapper children={<>ReactNode</>} className={"''"} />);
  expect(asFragment()).toMatchSnapshot();
});
