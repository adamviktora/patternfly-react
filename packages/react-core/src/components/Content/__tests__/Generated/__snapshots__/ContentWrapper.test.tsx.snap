// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentWrapper should match snapshot (auto-generated) 1`] = `
<DocumentFragment>
  <div
    class="pf-v6-c-content ''"
  >
    ReactNode
  </div>
</DocumentFragment>
`;
